### PR TITLE
New Risor options: WithoutGlobal, WithoutGlobals, WithGlobalOverride

### PR DESCRIPTION
**Pull request by myzie**
cc @luisdavim

### CODE REVIEW

File: risor.go [hunk 1]
```diff
@@ -33,6 +34,61 @@ func WithGlobal(name string, value any) Option {
...
+			return
+		}
+		// Find the named module
+		moduleName := parts[0]
+		var module *object.Module
+		if obj, ok := cfg.Globals[moduleName]; ok {
+			if m, ok := obj.(*object.Module); ok {
+				module = m
+			}
+		} else if obj, ok := cfg.DefaultGlobals[moduleName]; ok {
```
Comment (luisdavim) [inline risor.go:55]: if the same name is set on both `cfg.Globals` and `cfg.DefaultGlobals` it's enough to override/remove the first one? I think that works well for override but I'm not sure about the removal, but I'm also not sure if such edge case would even bee possible.

Comment (myzie) [reply risor.go]: Good call. It's possible.

**Comment by myzie** [comment]
@luisdavim - how does this look?

I could've added to your PR but I started this as an experiment and I didn't want to step on your work without running my changes by you.

File: risor.go [hunk 2]
```diff
@@ -33,6 +34,75 @@ func WithGlobal(name string, value any) Option {
...
+		if valueObj == nil || valueObj.Type() == object.ERROR {
+			panic(fmt.Sprintf("invalid value for global override: %v", value))
+		}
+		moduleName := parts[0]
+		nestedModulePath := parts[1 : len(parts)-1]
+		attrName := parts[len(parts)-1]
+		if obj, ok := cfg.Globals[moduleName]; ok {
+			if m, ok := obj.(*object.Module); ok {
+				if targetMod, ok := resolveModule(m, nestedModulePath); ok {
+					targetMod.SetAttr(attrName, valueObj)
```
Comment (luisdavim) [inline risor.go:92]: can't it be dangerous to allow doing this through `SetAttr`? that was why in my implementation I had a dedicated `Override` method that wasn't accessible from `risor` code directly.

**Comment by luisdavim** [comment]
> @luisdavim - how does this look?
> 
> I could've added to your PR but I started this as an experiment and I didn't want to step on your work without running my changes by you.

Yeah, no worries 👍 .

This LGTM but I'm not sure if allowing overriding module properties through `SetAttr` is the best option, I can see pros and cons, I had thought about that but ended up creating the separate `Override` method for it. Up to you, just something to think about....

File: risor.go (hunk 2 above)
Comment (myzie) [reply risor.go:92]: You're thinking of the situation when Risor is being used in a sandboxing-like capability, right? 

I'm doing a quick review of things from this point of view. Will add back `Override` depending on how this all looks.

Comment (luisdavim) [reply risor.go:92]: > You're thinking of the situation when Risor is being used in a sandboxing-like capability, right?

yes

Comment (myzie) [reply risor.go]: I've used OPA extensively, so this sounds familiar :-)

**Comment by myzie** [comment]
@luisdavim - I added Override back. LMK if there's anything off. But I merged already since it seems like this should do it for you.

**Pull request merged**